### PR TITLE
ServerTemplateSelector selection when there is only one ServerTemplat…

### PR DESCRIPTION
**Pull request by nmirasch**
…e available

Maintain selected the serverTemplate at selector when only exist one available.

### CODE REVIEW

**Comment by nmirasch** [comment]
@cristianonicolai can you take a look,please. It's  temporary and the idea is save us time when we are working with one kie-server-template, (waiting for persist it).

File: jbpm-console-ng-generic/jbpm-console-ng-generic-client/src/main/java/org/jbpm/console/ng/gc/client/menu/ServerTemplateSelectorMenuBuilder.java [hunk 1]
```diff
@@ -47,16 +47,31 @@
...
+        this.specManagementService =specManagementService;
+        this.view = view;
+    }
+
     @PostConstruct
     public void init() {
         specManagementService.call(new RemoteCallback<Collection<ServerTemplate>>() {
             @Override
             public void callback(final Collection<ServerTemplate> serverTemplates) {
+                String servletTemplateSelectedId = null;
```
Comment (cristianonicolai) [inline jbpm-console-ng-generic/jbpm-console-ng-generic-client/src/main/java/org/jbpm/console/ng/gc/client/menu/ServerTemplateSelectorMenuBuilder.java]: servlet? :)

File: jbpm-console-ng-generic/jbpm-console-ng-generic-client/src/main/java/org/jbpm/console/ng/gc/client/menu/ServerTemplateSelectorViewImpl.java [hunk 2]
```diff
@@ -65,6 +65,23 @@ public void init() {
     }
 
     @Override
+    public void selectServerTemplate(final String serverTemplateId) {
+        if (changeCommand != null) {
```
Comment (cristianonicolai) [inline jbpm-console-ng-generic/jbpm-console-ng-generic-client/src/main/java/org/jbpm/console/ng/gc/client/menu/ServerTemplateSelectorViewImpl.java:69]: I think you could try to somehow share this code with the AnchorListItem clickhandler from addServerTemplate, they are basically the same once you find the AnchorListItem you need.

File: jbpm-console-ng-generic/jbpm-console-ng-generic-client/src/main/java/org/jbpm/console/ng/gc/client/menu/ServerTemplateSelectorMenuBuilder.java [hunk 3]
```diff
@@ -47,16 +47,31 @@
     @Inject
     private Event<ServerTemplateSelected> serverTemplateSelectedEvent;
 
+
+    public ServerTemplateSelectorMenuBuilder(){
+
+    }
+
+    public ServerTemplateSelectorMenuBuilder(Caller<SpecManagementService> specManagementService, ServerTemplateSelectorView view ){
+        this.specManagementService =specManagementService;
```
Comment (cristianonicolai) [inline jbpm-console-ng-generic/jbpm-console-ng-generic-client/src/main/java/org/jbpm/console/ng/gc/client/menu/ServerTemplateSelectorMenuBuilder.java]: I started to use setter methods with `@Inject`. I found it to be more flexible than changing the constructor. Mockito autocratically injects the mocks it knows and for the caller you just use the setter method in the setup instead.

**Comment by nmirasch** [comment]
@cristianonicolai can you take a look on this? thanks!

**Comment by cristianonicolai** [comment]
+1, thanks @nmirasch

**Pull request merged**